frame/staking/src/lib.rs

```diff
@@ -1984,6 +1984,55 @@ decl_module! {
 			Self::do_payout_stakers(validator_stash, era)
 		}
 
+		/// Rebond a portion of the stash scheduled to be unlocked.
+		///
+		/// The dispatch origin must be signed by the controller, and it can be only called when
+		/// [`EraElectionStatus`] is `Closed`.
+		///
+		/// # <weight>
+		/// - Time complexity: O(L), where L is unlocking chunks
+		/// - Bounded by `MAX_UNLOCKING_CHUNKS`.
+		/// - Storage changes: Can't increase storage, only decrease it.
+		/// ---------------
+		/// - DB Weight:
+		///     - Reads: EraElectionStatus, Ledger, Locks, [Origin Account]
+		///     - Writes: [Origin Account], Locks, Ledger
+		/// # </weight>
+		#[weight = T::WeightInfo::rebond(MAX_UNLOCKING_CHUNKS as u32)]
+		fn rebond(
+			origin,
+			#[compact] plan_to_rebond_ring: RingBalance<T>,
+			#[compact] plan_to_rebond_kton: KtonBalance<T>
+		) -> DispatchResultWithPostInfo {
+			ensure!(Self::era_election_status().is_closed(), <Error<T>>::CallNotAllowed);
+
+			let controller = ensure_signed(origin)?;
+			let mut ledger = Self::ledger(&controller).ok_or(<Error<T>>::NotController)?;
+			let now = <frame_system::Module<T>>::block_number();
+
+			ledger.ring_staking_lock.update(now);
+			ledger.kton_staking_lock.update(now);
+
+			ensure!(
+				!ledger.ring_staking_lock.unbondings.is_empty()
+					|| !ledger.kton_staking_lock.unbondings.is_empty(),
+				<Error<T>>::NoUnlockChunk
+			);
+
+			ledger.rebond(plan_to_rebond_ring, plan_to_rebond_kton);
```

Comment:
Just to make sure this is as expected, can add a testcase for this:

The sum of tokens in lock.unbonding might not enough for the plan_to_rebond.

Reply:
If the unbonding is not enough, should we use free_balance, or throw an error, or keep as it is now?